src/Zypper.cc

```diff
@@ -1362,20 +1362,27 @@ void Zypper::processGlobalOptions()
 
   if ( (it = gopts.find("cache-dir")) != gopts.end() )
   {
+    ZConfig::instance().setRepoCachePath( it->second.front() );
     _gopts.rm_options.repoCachePath		= it->second.front();
     _gopts.rm_options.repoRawCachePath		= _gopts.rm_options.repoCachePath / "raw";
     _gopts.rm_options.repoSolvCachePath		= _gopts.rm_options.repoCachePath / "solv";
     _gopts.rm_options.repoPackagesCachePath	= _gopts.rm_options.repoCachePath / "packages";
   }
```

Comment:
In fact zypper should also not know about `"raw"`, `"solv"` and `"packages"`.  IMO ugly enough that zypper needs to know which dependent paths need to be reset, after `repoCachePath` has canged.
Shouldn't we assign `ZConfig::instance().repoMetadataPath()`, `repoSolvfilesPath()` and `repoPackagesPath()` instead?

Reply:
Right, then the logic for that is all in ZConfig